packages/subgraph/src/plugin/pluginRepo.ts

```diff
@@ -1,20 +1,60 @@
-import {VersionCreated} from '../../generated/templates/PluginRepoTemplate/PluginRepo';
-import {PluginVersion, PluginSetup} from '../../generated/schema';
+import {
+  ReleaseMetadataUpdated,
+  VersionCreated
+} from '../../generated/templates/PluginRepoTemplate/PluginRepo';
+import {
+  PluginVersion,
+  PluginSetup,
+  PluginRelease
+} from '../../generated/schema';
 
 export function handleVersionCreated(event: VersionCreated): void {
-  let id = `${event.address.toHexString()}_${event.params.release.toString()}_${event.params.build.toString()}`;
+  // PluginSetup
   let pluginSetupId = event.params.pluginSetup.toHexString();
-  let entity = new PluginVersion(id);
-  entity.pluginRepo = event.address.toHexString();
-  entity.release = event.params.release;
-  entity.build = event.params.build;
-  entity.pluginSetup = pluginSetupId;
-  entity.metadata = event.params.buildMetadata;
-  entity.save();
 
   let pluginSetupEntity = PluginSetup.load(pluginSetupId);
   if (!pluginSetupEntity) {
     pluginSetupEntity = new PluginSetup(pluginSetupId);
     pluginSetupEntity.save();
   }
+
+  // PluginVersion
+  let pluginRepoId = event.address.toHexString();
+  let pluginRelease = event.params.release.toString();
+  let pluginReleaseId = pluginRepoId.concat('_').concat(pluginRelease);
+  let pluginVersionId = pluginReleaseId
+    .concat('_')
+    .concat(event.params.build.toString());
+
+  let entity = new PluginVersion(pluginVersionId);
+  entity.pluginRepo = event.address.toHexString();
+  entity.pluginSetup = pluginSetupId;
+
+  entity.release = pluginReleaseId;
+  entity.build = event.params.build;
+
+  entity.metadata = event.params.buildMetadata.toString();
+  entity.save();
+}
+
+export function handleReleaseMetadataUpdated(
+  event: ReleaseMetadataUpdated
+): void {
+  let pluginRepoId = event.address.toHexString();
+  let pluginRelease = event.params.release;
+  let releaseMetadata = event.params.releaseMetadata.toString();
+
+  let pluginReleaseEntityId = pluginRepoId
+    .concat('_')
+    .concat(pluginRelease.toString());
+
+  let pluginReleaseEntity = PluginRelease.load(pluginReleaseEntityId);
```

Comment:
no need to load first and then calling `new`. As I found out subgraph merges the entities if you create a new one and the same exists already 
https://thegraph.com/docs/en/developing/assemblyscript-api/#updating-existing-entities

Reply:
nice, since we use load everywhere i guess we can go with load.